mobile/lib/components/utility/image_with_overlay.dart

```diff
@@ -0,0 +1,85 @@
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+import 'package:radioaktywne/extensions/build_context.dart';
+
+/// Widget representing an image with overlaid text on the bottom of the image.
+class ImageWithOverlay extends StatelessWidget {
+  const ImageWithOverlay({
+    super.key,
+    this.child,
+    required this.thumbnailPath,
+    this.imageConstructor = Image.network,
+    this.titleOverlay,
+    required this.isLoading,
+    this.titleOverlayPadding = const EdgeInsets.all(4),
+  });
+
+  /// Child widget to display centered above the image, BUT ABOVE TITLE OVERLAY.
+  final Widget? child;
+
+  /// Network or local path to the image.
+  final String thumbnailPath;
+
+  /// Function used to display the image (used mainly to be able to switch
+  /// between network and local images). Defaults to network image.
+  final Image Function(String, {BoxFit fit}) imageConstructor;
+
+  /// Title to display over the image, on the bottom.
+  final Widget? titleOverlay;
+
+  /// Boolean value used to display default image when image hasn't loaded yet.
+  final bool isLoading;
+
+  /// lets optionally change padding around the title.
+  final EdgeInsets titleOverlayPadding;
+
+  @override
+  Widget build(BuildContext context) {
+    return Stack(
+      children: <Widget>[
+        Positioned.fill(
+          child: Container(
+            color: context.colors.backgroundDarkSecondary,
+            child: isLoading
+                ? Image.asset(
+                    'assets/defaultMedia.png',
+                    fit: BoxFit.cover,
+                  )
+                : imageConstructor(
+                    thumbnailPath,
+                    fit: BoxFit.cover,
+                  ),
```

Comment:
We would have to ask RA on this, but I think there are 3 states at least - loading, success and fail. I think you should let the user handle all 3 (because it is really easy thanks to `Image.<something>()`'s optional builder methods).
tl;dr: would recommend to just leave it as an image with no state and let the user handle those states.

Reply:
I managed to rewrite it with errorBuilder and loadingBuilder in mind, have a look and see if that's what it's supposed to look like.